### ISSUE TITLE
Summary in Reports

### ISSUE THREAD

**Issue by Jeeppler** [mercedes-benz/sechub#345]
It would be very helpful to have a summary in the report. Currently, there is no summary of the findings found. Having a summary in the report would allow a human to get a quick overview of the findings in the report, without having to manually count the number of findings. Counting the number of findings is time consuming, especially in large projects with many findings.

Furthermore, SecHub reports should be readable without requiring any additional tools. Including a summary in the JSON and HTML report would allow that.

Most important to know is the number of findings per scan type and how many findings red, yellow and green findings were found. For example:

Code Scan:
- Total: 18
- Red: 0
- Yellow: 1
- Green: 17

Infra Scan:
- Total: 11
- Red: 0
- Yellow: 4
- Green: 7

In addition, the report could include a more detailed overview of the finding categories. For example:

Code Scan

Medium: 
- CWE-36 - Absolute Path Traversal (Count: 1)

Low:
- CWE-379 - Creation of Tempdir... (Count: 5)
- CWE-259 - Use Of Hardcoded Password (Count: 3)

The mock-up below shows how a summary could look like in the HTML report.

![sechub_summary_report_proposal](https://user-images.githubusercontent.com/2007739/90923717-71d92800-e3ee-11ea-8a8b-7641a48a269a.png)

The following JSON shows the how a summary could look like in the JSON report.

~~~
{
   "jobUUID": "b2832de0-b9ce-4daa-b7b8-5fd89b9f62c4",
   "result": {
      "count": 27,
      "metadata": {
        "summary": {
            "codeScan": {
                "total": 18,
                "red": 0,
                "yellow": 1,
                "green": 17,
                "details": {
                    "medium": [
                        {
                            "cweId": 36,
                            "name": "CWE-36 - Absolute Path Traversal",
                            "count": 1
                        }
                    ],
                    "low": [
                        {
                            "cweId": 379,
                            "name": "CWE-379 - Creation of Tempdir...",
                            "count": 5, 
                        }
                        {   
                            "cweId": 259,
                            "name": "Use Of Hardcoded Password",
                            "count": 3, 
                        }
                    ]
                }
            },
            "infraScan": {
                "total": 11,
                "red": 0,
                "yellow": 4,
                "green": 7,
                "details": {
                    ...
                }
            },
        }
      },
      "findings": [
         {
            "id": 220,
            "name": "Portability Flaw In File Separator",
            "severity": "INFO",
            "code": {
               "location": "handlers/Authentication.java",
               "line": 162,
               "column": 69,
          ...
~~~

---
<sup>Jeremias Eppler <jeremias.eppler@daimler.com>, Daimler TSS GmbH, [imprint](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)</sup>

**Comment by sven-dmlr** [comment]
Good Ideas :+1: 

From a reporting perspective, it would be helpful if the summary data is easily accessible in the database. (not only stored as a large text field)

**Comment by Jeeppler** [comment]
@sven-dmlr it should be stored in the database to be able to do reporting.

**Comment by Jeeppler** [comment]
It would be nice to have the summary in the JSON report as well. The JSON report is the default report and it is tedious to analyze the JSON report by the "search" feature.

**Comment by zigfridus** [comment]
@Jeeppler 
assign me to this issue please.

**Comment by Jeeppler** [comment]
@zigfridus I can confirm that is a bug. I tried to replicate it and get an empty report back as well.

**Comment by Jeeppler** [comment]
I created #2071 to track the problem.

**Comment by Jeeppler** [comment]
Another way to test it: start a pds-solution, like PMD, in a container and connect SecHub to it. 

~~~
[ SecHub in your IDE ]  <-----> [ PMD+PDS ]
~~~

Next, you have to setup a project, executor etc. You can find scripts in `sechub-solution/setup-pds` to help you with the setup. For PMD the script is called: `setup-pmd.sh`. You have to adjust the default values to the integration test mode.

JVM arguments for the integration test mode:

~~~
-Dspring.profiles.active=mocked_products,h2,integrationtest
-Dsechub.server.debug=true
-Dsechub.storage.sharedvolume.upload.dir=/home/user/.sechub/sharedvolume
-Dsechub.targettype.detection.intranet.hostname.endswith=intranet.example.org
-Dsechub.config.trigger.nextjob.initialdelay=0
~~~

**Comment by Jeeppler** [comment]
In both cases you have to start only SecHub in the IDE.

Option A) Start SecHub in integration test mode -> get a mock report back. Problem, currently it does not work: #2071.
Option B) Start SecHub with the `real_products` profile and start a PDS-Solution. Configure SecHub to use the PDS-solution with the help of scripts in `sechub-solution/setup-pds` -> scan real projects and get real results back.

**Comment by zigfridus** [comment]
@Jeeppler 
There are multiple tasks in this issue.  The tasks from you, @de-jcup and @sven-dmlr.  I would like to ask you to streamline these tasks and specify with what task it's better to start.
Thanks

**Comment by Jeeppler** [comment]
@zigfridus I will refine it with @de-jcup, @sven-dmlr.

**Comment by de-jcup** [comment]
We should fix the mentioned parts. But this issue can be done easier and faster with next steps described:

### Report development without server starts

_Remark @zigfridus  The next step  will work after #2124  has been merged  (we had this already, but there was bug #2125 which is fixed by this merge)_ 

#### Development changes in HTML report
- Change CSS inside `scanresult.css`
- Change Template content in `scanresult.html` and included fragements

#### Generate the output
- Just  execute `ThymeLeafHTMLReportingTest`  with  
   - ENV variable `SECHUB_KEEP_TEMPFILES` to `true`  
   - ENV variable `SECHUB_AUTO_GENERATE_CSS_FRAGMENTS_ON_HTML_TESTS` to `true`

#### Check
Look at the output file paths in console and open one of the mentioned html files with your browser and inspect the result. The path will not change, so after changing + restarting unit tests you can simply refresh your browser tab.

This should increase the report development very much.

**Comment by zigfridus** [comment]
@Jeeppler 
Thank you for the detailed explanation.

**Comment by zigfridus** [comment]
@Jeeppler 
Where I can download a source code with Medium and High severity of vulnerability for testing?
Thanks

**Comment by zigfridus** [comment]
@Jeeppler
I've found the solution already. Sorry for disturbing you.

**Comment by zigfridus** [comment]
@Jeeppler
I created draft PR. It only creates Summary table in html.  Please check if this is what you expected to see. Thanks

**Comment by Jeeppler** [comment]
@zigfridus I reviewd it.

**Comment by zigfridus** [comment]
@Jeeppler 
Thank you

**Comment by dAnjou** [comment]
Hi, it'd be nice to have info in the report that indicates what has been scanned. Currently, the report for a successful job is quite empty, and you can't tell whether that's because there are actually no findings or whether nothing has been scanned:
```json
{
   "result": {
      "count": 0,
      "findings": []
   },
   "reportVersion": "1.0",
   "jobUUID": "930ea58e-f220-437f-a468-0046543c1573",
   "trafficLight": "GREEN",
   "messages": [],
   "status": "SUCCESS"
}
```
![Screenshot 2023-06-20 at 18 28 57](https://github.com/mercedes-benz/sechub/assets/117147/fb699aa6-7eba-492a-8e39-f39885ec20ed)

**Comment by Jeeppler** [comment]
@dAnjou thanks. However, your comment does not belong in this issue. Please create a new issue or get in contact with @sven-dmlr or @haerter-tss.

**Comment by de-jcup** [comment]
Attached some examples how the final reports (HTML/JSON) looks like.

[sechub-report-examples.zip](https://github.com/mercedes-benz/sechub/files/14310832/sechub-report-examples.zip)
